WalletWasabi.Tests/UnitTests/GUI/CommandInterpreterTest.cs

```diff
@@ -0,0 +1,42 @@
+using Mono.Options;
+using System.IO;
+using WalletWasabi.Gui.CommandLine;
+using WalletWasabi.Helpers;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.GUI
+{
+	public class CommandInterpreterTest
+	{
+		[Fact]
+		public async void CommandInterpreterShowsHelpAsync()
+		{
+			var textWriter = new StringWriter();
+			var c = new CommandInterpreter(textWriter);
+			await c.ExecuteCommandsAsync(new string[] { "wassabee", "--help" }, new Command("mixer"), new Command("findpassword"));
```

Comment:
This test is wrong. It asserts Wasabi return the incorrect output. I suspect this line is wrong and should be:
```suggestion
			await c.ExecuteCommandsAsync(new string[] {"--help" }, new Command("mixer"), new Command("findpassword"));
```

Make sure to assert the correct outputs for the different commands.

Reply:
No, the suggested change does not work at all. Please elaborate what incorrect output you mean.

Reply:
Lets first agree that the test is wrong because wasabi doesn't display what the test asserts. It seems that instead of asserting what the code should display this test asserts whatever the code produced.

Now, there are a couple of this that are not correct:
 
1) In dotnet the `string[] args` do not contain the name of the executable. This means that if you pass `new string[] { "wassabee", "--help" }` then `wassabee` is the first argument and `--help` is the second argument. This is equivalent to do this from the terminal:

```
$ wassabee wassabee --help
```

So, I didn't test my suggestion but it is clearly correct and you have to remove the first `wassabee`.

 2) When you invoke `wassabee --help` it is expected to display the short help. It should not display the protocol version if I remember correctly. The version should be displayed when you invoke it with `wassabee --version` only.

Reply:
@lontivero Please see the new unit tests. You are right, but even that ` wassabee wassabee --help` is actually a thing (see `InvalidCommandWithHelpAsync`). The code is actually quite complex.